packages/next/src/server/next.ts

```diff
@@ -308,7 +309,14 @@ export class NextServer implements NextWrapperServer {
   private async getServer() {
     if (!this.serverPromise) {
       this.serverPromise = this[SYMBOL_LOAD_CONFIG]().then(async (conf) => {
-        if (!this.options.dev) {
+        if (this.options.dev) {
+          if (conf.experimental.lockDistDir) {
+            await Lockfile.acquireOrExit(
+              path.join(conf.distDir, 'lock'),
+              'next dev'
```

Comment:
probably just `next`?  or we need two lock files?

Reply:
`next dev` and `next build` now have separate `distDir`s by default: https://github.com/vercel/next.js/pull/83961